pkg/operator/ceph/cluster/osd/status.go

```diff
@@ -166,6 +166,11 @@ func (c *Cluster) completeOSDsForAllNodes(config *provisionConfig, configOSDs bo
 
 	currentTimeoutMinutes := 0
 	for {
+		// Check whether we need to cancel the orchestration
+		if c.context.RequestCancelOrchestration.IsSet() {
```

Comment:
couldn't use `CheckForCancelledOrchestration(c.context)` because its `unsets` the orchestration cancellation.

Reply:
I believe we want to follow the pattern that is set in other places to add an error and then return such as:
```go
		if err := controller.CheckForCancelledOrchestration(c.context); err != nil {
			config.addError("%s", err.Error())
			return
		}
```
Or do you see a different pattern here?

Reply:
I did try this. `completeOSDsForAllNodes` is used by below methods in the same order.

 1. `startProvisioningOverPVCs`

 2. `startProvisioningOverNodes`


`CheckForCancelledOrchestration` also `unsets` the `RequestCancelOrchestration`.  Because of this, `startProvisioningOverPVCs` will work. But when `startProvisioningOverNodes` is executed, 
it will again wait for 20 miniutes as the `RequestCancelOrchestration` is unset.

Reply:
I don't understand why that https://github.com/rook/rook/blob/ba91f5d7f8742c9b1d47874059bbcfe8caff488f/pkg/operator/ceph/controller/controller_utils.go#L55 is an issue.

The cancellation is set only when the CR is deleted or the image in the CR is updated. If you use `CheckForCancelledOrchestration()`, it will always be atomic, or is this code checking in 2 places at the same time?

Reply:
What if we skip `startProvisioningOverNodes()` if there are any errors? Something like: `if len(config.errorMessages) > 0`. Clusters will most commonly only be using one or the other type of OSDs, so it seems reasonable to skip the latter if the first one has errors.

Reply:
This works. Updated the code.